src/cli/index.js

```diff
@@ -398,14 +399,32 @@ export function main({
     });
 }
 
-export default function start() {
-  // ignore all arguments after a --
-  const doubleDashIndex = process.argv.findIndex(element => element === '--');
-  const startArgs = process.argv.slice(0, 2);
-  const args = process.argv.slice(2, doubleDashIndex === -1 ? process.argv.length : doubleDashIndex);
-  const endArgs = doubleDashIndex === -1 ? [] : process.argv.slice(doubleDashIndex + 1, process.argv.length);
+async function start(): Promise<void> {
+  const rc = getRcConfigForCwd(process.cwd());
+  const yarnPath = rc['yarn-path'];
 
-  main({startArgs, args, endArgs});
+  if (yarnPath && process.env.YARN_IGNORE_PATH !== '1') {
+    const argv = process.argv.slice(2);
+    const opts = {stdio: 'inherit', env: Object.assign({}, process.env, {YARN_IGNORE_PATH: 1})};
+
+    try {
+      await spawnp(yarnPath, argv, opts);
+    } catch (firstError) {
+      try {
+        await forkp(yarnPath, argv, opts);
+      } catch (error) {
+        throw firstError;
+      }
+    }
+  } else {
+    // ignore all arguments after a --
+    const doubleDashIndex = process.argv.findIndex(element => element === '--');
+    const startArgs = process.argv.slice(0, 2);
+    const args = process.argv.slice(2, doubleDashIndex === -1 ? process.argv.length : doubleDashIndex);
+    const endArgs = doubleDashIndex === -1 ? [] : process.argv.slice(doubleDashIndex + 1, process.argv.length);
```

Comment:
Nope, this is the same code as the one removed [here](https://github.com/yarnpkg/yarn/pull/4204/files/890bab1bd8e0767704bf29bb1b908d500369a10c#diff-867becf4a9c2c6c6d4e7c1278750724eL402), except that it now is in an `else` branch.